src/features/cellnav/js/cellnav.js

```diff
@@ -641,132 +648,153 @@
           return {
             pre: function ($scope, $elm, $attrs, uiGridCtrl) {
               var _scope = $scope;
+              var api;
 
               var grid = uiGridCtrl.grid;
-              uiGridCellNavService.initializeGrid(grid);
-
-              uiGridCtrl.cellNav = {};
-
-              //Ensure that the object has all of the methods we expect it to
-              uiGridCtrl.cellNav.makeRowCol = function (obj) {
-                if (!(obj instanceof GridRowColumn)) {
-                  obj = new GridRowColumn(obj.row, obj.col);
-                }
-                return obj;
-              };
-
-              uiGridCtrl.cellNav.getActiveCell = function () {
-                var elms = $elm[0].getElementsByClassName('ui-grid-cell-focus');
-                if (elms.length > 0){
-                  return elms[0];
-                }
-
-                return undefined;
-              };
-
-              uiGridCtrl.cellNav.broadcastCellNav = grid.cellNav.broadcastCellNav = function (newRowCol, modifierDown, originEvt) {
-                modifierDown = !(modifierDown === undefined || !modifierDown);
-
-                newRowCol = uiGridCtrl.cellNav.makeRowCol(newRowCol);
-
-                uiGridCtrl.cellNav.broadcastFocus(newRowCol, modifierDown, originEvt);
-                _scope.$broadcast(uiGridCellNavConstants.CELL_NAV_EVENT, newRowCol, modifierDown, originEvt);
-              };
-
-              uiGridCtrl.cellNav.clearFocus = grid.cellNav.clearFocus = function () {
-                grid.cellNav.focusedCells = [];
-                _scope.$broadcast(uiGridCellNavConstants.CELL_NAV_EVENT);
-              };
-
-              uiGridCtrl.cellNav.broadcastFocus = function (rowCol, modifierDown, originEvt) {
-                modifierDown = !(modifierDown === undefined || !modifierDown);
-
-                rowCol = uiGridCtrl.cellNav.makeRowCol(rowCol);
 
-                var row = rowCol.row,
-                  col = rowCol.col;
+              function setupCellNav() {
```

Comment:
I'm apparently really bad at indentation and stuff, but you wrapped all of this stuff in setupCellNav, right?

Reply:
Basically yeah. Most of this stuff has just been moved into this function so we can toggle the behavior